tasks/main.yml

```diff
@@ -45,18 +45,26 @@
 
     - name: Ensure Last modified header is absent
       lineinfile:
-        dest: /etc/postfix/main.cf
+        path: /etc/postfix/main.cf
         regexp: '# Last modified:'
         state: absent
 
-    - name: Ensure Ansible Managed header in configuration file
+    # Previously, the role inserted a plain-text comment at the top of main.cf.
+    # This task removes this outdated header for compatibility.
+    - name: Ensure the outdated ansible managed header is absent
       lineinfile:
-        dest: /etc/postfix/main.cf
-        regexp: 'managed by [aA]nsible'
-        state: present
+        path: /etc/postfix/main.cf
+        regexp: "# This file is managed by [aA]nsible"
+        state: absent
+
+    - name: Ensure ansible_managed header in configuration file
+      vars:
+        __postfix_ansible_managed: "{{
+          lookup('template', 'get_ansible_managed.j2') }}"
+      blockinfile:
+        path: /etc/postfix/main.cf
+        block: "{{ __postfix_ansible_managed }}"
```

Comment:
@spetrosi, a question came up as I was going to borrow this code for mssql. ;)
Does this `blockinfile` add `__postfix_ansible_managed` every time  when this `Ensure ansible_managed header in configuration file` task is executed?
1) when `/etc/postfix/main.cf` is bland new (without `# This file is managed by [aA]nsible`), it's good.
2) when `/etc/postfix/main.cf` already has the same header as `__postfix_ansible_managed` contains, `blockinfile` does nothing, I guess. Then, it's good.
3) when `/etc/postfix/main.cf` already has a different header from what `__postfix_ansible_managed` contains, can the existing header be replaced with the new one? It's not so clear to me...

The kernel_settings role captures the existing header and compares it with the new one. Based upon the result, it determines to add, replace or skip [0]. Do you think it is needed for `mssql`? How about here?
[0] - https://github.com/linux-system-roles/kernel_settings/blob/master/tasks/main.yml#L48-L102

Reply:
try writing a simple test using `blockinfile` to see if it changes the header if different, and is idempotent - it is supposed to be (like all ansible modules) - also take a look at the source code for `blockinfile`

Reply:
Sorry, I should have... :p
Basically, `blockinfile` adds its own BEGIN and END and manage the strings in between. So, my concerns are all cleared. Thanks!
```
# BEGIN ANSIBLE MANAGED BLOCK
#
# New ansible managed
# Updated by nhosoi
#
# END ANSIBLE MANAGED BLOCK
[test]
abc: xyz
```